### PR TITLE
PIM-5579: Add the possibility to disable the export of products medias if needed

### PR DESCRIPTION
**Pull request by phaseinducer**
| Q                 | A
| ----------------- | ---
| Specs             | n/a
| Behats            | n/a
| Blue CI           | 
| Changelog updated | yes
| Review and 2 GTM  |

PR for doc : https://github.com/akeneo/pim-docs/pull/237

### CODE REVIEW

File: src/Pim/Bundle/BaseConnectorBundle/Writer/File/CsvProductWriter.php [hunk 1]
```diff
@@ -19,6 +19,9 @@ class CsvProductWriter extends ContextableCsvWriter
     /** @var FileExporterInterface */
     protected $fileExporter;
 
+    /** @var bool */
+    protected $copyMedias;
```
Comment (solivier) [inline src/Pim/Bundle/BaseConnectorBundle/Writer/File/CsvProductWriter.php]: copyMedia*

File: CHANGELOG-1.4.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
+# 1.4.x
+
+## Scalability improvements
+- PIM-5579: Add the possibility to disable the export of products medias if needed
```
Comment (solivier) [inline CHANGELOG-1.4.md]: media

**Comment by solivier** [comment]
gtm

**Comment by pierallard** [comment]
GTM

**Comment by jjanvier** [comment]
This fix really emphases the architecture problem around the exports and the archivers. It's a terrible hack... Basically you have a class that implements an archiver behavior and you bypass it with a boolean argument :(. I really don't like but I don't know how you could do export in a patch :(

GTM, with sadness

File: src/Pim/Bundle/BaseConnectorBundle/Writer/File/CsvProductWriter.php [hunk 3]
```diff
@@ -27,12 +30,14 @@ class CsvProductWriter extends ContextableCsvWriter
 
     /**
      * @param FileExporterInterface $fileExporter
+     * @param bool                  $copyMedia
      */
-    public function __construct(FileExporterInterface $fileExporter)
+    public function __construct(FileExporterInterface $fileExporter, $copyMedia = true)
```
Comment (skeleton) [inline src/Pim/Bundle/BaseConnectorBundle/Writer/File/CsvProductWriter.php]: why this param is not added in DI ?

**Comment by phaseinducer** [comment]
Agree. We can do much better for this whole archiving stuff, in 1.6 I hope.

File: src/Pim/Bundle/BaseConnectorBundle/Writer/File/CsvProductWriter.php (hunk 3 above)
Comment (phaseinducer) [inline src/Pim/Bundle/BaseConnectorBundle/Writer/File/CsvProductWriter.php]: Is it useful since it has a default value ?

Comment (ClementGautier) [inline src/Pim/Bundle/BaseConnectorBundle/Writer/File/CsvProductWriter.php]: Boolean flags are, by definition, bad solutions: http://phpmd.org/rules/cleancode.html#BooleanArgumentFlag.
I would suggest to create an other writer (CsvProductWithoutMediaWriter or something like this).

By the way, could we do that at a lower level ? I mean before the writer is called ? At processor level or reader level ? This way we could even not hydrate Product medias probably.

**Comment by ClementGautier** [comment]
missing specs

**Comment by BitOne** [comment]
Why disabling the product media export while the problem lies in the archiver, not in the export ?
I would do a parameter to disable the archiver (because this is the problem), not the product export.
If you have people that have the problem with the archiver but uses the standard media export to handle their media (which is the normal cases), this will not fix their problem: it will either break their export or not this the archiver...

**Comment by BitOne** [comment]
Not GTM

**Comment by phaseinducer** [comment]
Replaced by https://github.com/akeneo/pim-community-dev/pull/4028

**Pull request closed without merging**